src/Driver/WpcliDriver.php

```diff
@@ -104,41 +105,43 @@ public function wpcli($command, $subcommand, $raw_arguments = [])
 
         // Build parameter list.
         foreach ($raw_arguments as $name => $value) {
-            if (is_int($name)) {
+            if (is_numeric($name)) {
                 $arguments .= "{$value} ";
             } else {
                 $arguments .= sprintf('%s=%s ', $name, escapeshellarg($value));
             }
         }
 
-        // Support WP-CLI environment alias, or path and URL.
+        // TODO: review best practice with escapeshellcmd() here, and impact on metacharacters.
+        $config = sprintf('--path=%s --url=%s', escapeshellarg($this->path), escapeshellarg($this->url));
+
+        // Support WP-CLI environment aliases.
         if ($this->alias) {
             $config = "@{$this->alias}";
-        } else {
-            // TODO: review best practice with escapeshellcmd() here, and impact on metacharactes.
-            $config = sprintf('--path=%s --url=%s', escapeshellarg($this->path), escapeshellarg($this->url));
         }
 
 
-        $wpcli_args = sprintf('--no-color --require=%1$s/WpcliLogger.php', dirname(__DIR__));
-
         // Query WP-CLI.
-        $proc = proc_open("{$this->binary} {$config} {$wpcli_args} {$command} {$subcommand} {$arguments}", [
-            1 => ['pipe','w'],
-            2 => ['pipe','w'],
-        ], $pipes);
+        $proc = proc_open(
+            "{$this->binary} {$config} --no-color {$command} {$subcommand} {$arguments}",
+            array(
+                1 => ['pipe', 'w'],
+                2 => ['pipe', 'w'],
+            ),
+            $pipes
+        );
 
-        $cmd_output = stream_get_contents($pipes[1]);
+        $stdout = stream_get_contents($pipes[1]);
         fclose($pipes[1]);
         $stderr = stream_get_contents($pipes[2]);
         fclose($pipes[2]);
         $exit_code = proc_close($proc);
 
-        if ($exit_code || $stderr) {
+        if ($exit_code || $stderr || strpos($stdout, 'Warning: ') !== false || strpos($stdout, 'Error: ') !== false) {
             throw new UnexpectedValueException("WP-CLI driver query failure: {$stderr} ($exit_code)");
         }
 
-        return compact('cmd_output', 'exit_code');
+        return compact('stdout', 'stderr', 'exit_code');
     }
```

Comment:
Why include `stderr`? If it's non-empty then we don't reach this line.

If `stderr` is empty and `stdout` contains `Warning:` then we should probably include that rather than `stderr`.

Lastly, should we remove the `WP-CLI driver query failure`? It doesn't add anything...